sn_client/src/files/mod.rs

```diff
@@ -175,7 +181,12 @@ impl Files {
 
         let result = self.upload(chunks).await;
 
-        // drop the sender
+        // send an event indicating that the upload process completed with an error
+        if result.is_err() {
+            self.send_event(FileUploadEvent::Error).await?;
```

Comment:
Can we pass out the error here?

Reply:
The error variants inside ClientEvent does not impl Copy. We might have to wrap the internal types into `Arc` or return an arc error. There does not seem to be a simple workaround for it right now.